spark-connect/client/src/test/scala-spark-master/io/delta/connect/tables/DeltaTableBuilderSuite.scala

```diff
@@ -0,0 +1,533 @@
+  private val defaultTestTableProperty = Map("foo" -> "bar")
+  private val defaultTestTableClusteringColumns = Seq("c1", "c3")
+
+
+  /**
+   * Verify if the table metadata matches the test table. We use this to verify DDLs
+   * write correct table metadata into the transaction logs.
+   * If clusteringCols are not empty, it verifies the clustering columns for liquid.
+   */
+  protected def verifyTestTableMetadata(
+      table: String,
+      schemaString: String,
+      generatedColumns: Map[String, String] = Map.empty,
+      identityColumns: Map[String, String] = Map.empty,
+      colComments: Map[String, String] = Map.empty,
+      colNullables: Set[String] = Set.empty,
+      tableComment: Option[String] = None,
+      expectedPartitionCols: Seq[String] = Seq.empty,
+      expectedTableProperties: Map[String, String] = Map.empty,
+      expectedClusteringCols: Seq[String] = Seq.empty): Unit = {
+    val session = spark
+    import session.implicits._
+
+    val expectedSchema = StructType(StructType.fromDDL(schemaString).map { field =>
+      val newMetadata = new MetadataBuilder().withMetadata(field.metadata)
+      if (colComments.contains(field.name)) {
+        newMetadata.putString("comment", colComments(field.name))
+      }
+      field.copy(
+        nullable = colNullables.contains(field.name),
+        metadata = newMetadata.build())
+    })
+
+    val deltaTable = DeltaTable.forName(spark, table)
+    assert(deltaTable.toDF.schema == expectedSchema)
+
+    val (description, partitionColumns, properties, clusteringColumns) =
+      deltaTable.detail()
+        .select("description", "partitionColumns", "properties", "clusteringColumns")
+        .as[(String, Seq[String], Map[String, String], Seq[String])]
+        .head()
+
+    assert(description == tableComment.orNull)
+    assert(partitionColumns == expectedPartitionCols)
+    // It may contain other properties other than the ones we added.
+    expectedTableProperties.foreach {
+      prop => properties.contains(prop._1) && properties.get(prop._1).get == prop._2
+    }
+    assert(clusteringColumns == expectedClusteringCols)
+
+    val schemaFields = deltaTable.toDF.schema.fields
+
+    // Verify generated columns
+    generatedColumns.foreach { case (col, expr) =>
+      val fieldOpt = schemaFields.find(_.name == col)
+      assert(fieldOpt.isDefined, s"Generated column $col not found in table schema")
+      val field = fieldOpt.get
+      
+      // Check if the metadata contains the generation expression key
+      if (field.metadata.contains("delta.generationExpression")) {
+        val generationExpr = field.metadata.getString("delta.generationExpression")
+        assert(generationExpr == expr, 
+          s"Generated column $col has expression '$generationExpr' but expected '$expr'")
+      }
+    }
+    
+    // Verify identity columns without detailed metadata validation
+    identityColumns.foreach { case (col, expectedIdentityDef) =>
```

Comment:
```suggestion
    identityColumns.foreach { case (col, _) =>
```

Since we are not using it